opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1458,16 +1469,25 @@ namespace Opm
             segment_viscosities_[seg] = 0.;
             // calculate the average viscosity
             for (int comp_idx = 0; comp_idx < num_components_; ++comp_idx) {
-                const EvalWell comp_fraction = mix[comp_idx] / b[comp_idx] / volrat;
-                segment_viscosities_[seg] += visc[comp_idx] * comp_fraction;
+                const EvalWell fraction =  mix[comp_idx] / b[comp_idx] / volrat;
+                segment_phase_fractions_[seg][comp_idx] = fraction >= 0.0 ? fraction : 0.0;
```

Comment:
Some MSW tests fail because this line. Which means in the master branch, sometimes we use negative fractions, although very small. 

Anyway, the change should only make good, at least, no harm.

Reply:
This change impacts the running of one of the targeting cases.  Reverting it now and will introduce a little more sophisticated chopping.